oneandone/src/main/java/org/apache/jclouds/oneandone/rest/domain/LoadBalancer.java

```diff
@@ -0,0 +1,323 @@
+   public abstract static class CreateLoadBalancer {
+
+      public abstract String name();
+
+      @Nullable
+      public abstract String description();
+
+      public abstract HealthCheckTestTypes healthCheckTest();
+
+      public abstract int healthCheckInterval();
+
+      @Nullable
+      public abstract String healthCheckPath();
+
+      @Nullable
+      public abstract String healthCheckParse();
+
+      public abstract Boolean persistence();
+
+      @Nullable
+      public abstract Integer persistenceTime();
+
+      public abstract Types.LoadBalancerMethod method();
+
+      @Nullable
+      public abstract String datacenterId();
+
+      public abstract List<Rule.CreatePayload> rules();
+
+      @SerializedNames({"name", "description", "health_check_test", "health_check_interval", "health_check_path", "health_check_parse", "persistence", "persistence_time", "method", "datacenter_id", "rules"})
+      public static CreateLoadBalancer create(final String name, final String description, final HealthCheckTestTypes healthCheckTest, final int healthCheckInterval, final String healthCheckPath,
+              final String healthCheckParse, final Boolean persistence, final Integer persistenceTime, final Types.LoadBalancerMethod method, final String dataCenterId, final List<Rule.CreatePayload> rules) {
+         return builder()
+                 .name(name)
+                 .description(description)
+                 .healthCheckTest(healthCheckTest)
+                 .healthCheckInterval(healthCheckInterval)
+                 .healthCheckPath(healthCheckPath)
+                 .healthCheckParse(healthCheckParse)
+                 .persistence(persistence)
+                 .persistenceTime(persistenceTime)
+                 .method(method)
+                 .datacenterId(dataCenterId)
+                 .rules(rules == null ? ImmutableList.<Rule.CreatePayload>of() : ImmutableList.copyOf(rules))
+                 .build();
+      }
+
+      public static Builder builder() {
+         return new AutoValue_LoadBalancer_CreateLoadBalancer.Builder();
+      }
+
+      @AutoValue.Builder
+      public abstract static class Builder {
+
+         public abstract Builder name(String name);
+
+         public abstract Builder description(String description);
+
+         public abstract Builder healthCheckTest(HealthCheckTestTypes healthCheckTest);
+
+         public abstract Builder healthCheckInterval(int healthCheckInterval);
+
+         public abstract Builder healthCheckPath(String healthCheckPath);
+
+         public abstract Builder healthCheckParse(String healthCheckParse);
+
+         public abstract Builder persistence(Boolean persistence);
+
+         public abstract Builder persistenceTime(Integer persistenceTime);
+
+         public abstract Builder method(Types.LoadBalancerMethod method);
+
+         public abstract Builder datacenterId(String datacenterId);
+
+         public abstract Builder rules(List<Rule.CreatePayload> rules);
```

Comment:
Sure. Take a look at [this builder](https://github.com/jclouds/jclouds-labs/blob/3be36289f594c42af2f46daee1661556fb0e3a35/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/NetworkProfile.java#L42-L54). The trick consists on providing a hidden accessor for the collection in the builder, and a hidden build method (we usually call it `autoBuild()`). The we provide a non-abstract build method that uses the hidden one (implemented by Google Auto) and enforces immutability in all fields that require it.